# LAB00 — Verificación del entorno

**INF-8239 Ciencia de Datos II · Unidad 01 · Ejercicio 01**

Este notebook verifica que el entorno esté correctamente configurado antes de comenzar. Se puede ejecutar de principio a fin con *Kernel nuevo → Run All*.

## 1. Versiones de Python y bibliotecas

In [1]:
import sys
import platform

print('Sistema operativo:', platform.system(), platform.release())
print('Python:', sys.version)

Sistema operativo: Windows 11
Python: 3.13.5 (tags/v3.13.5:6cb20a2, Jun 11 2025, 16:15:46) [MSC v.1943 64 bit (AMD64)]


In [2]:
import numpy as np
import pandas as pd
import sklearn
import matplotlib
import seaborn as sns
import joblib

print('NumPy:', np.__version__)
print('pandas:', pd.__version__)
print('scikit-learn:', sklearn.__version__)
print('matplotlib:', matplotlib.__version__)
print('seaborn:', sns.__version__)
print('joblib:', joblib.__version__)

NumPy: 2.3.2
pandas: 2.3.1
scikit-learn: 1.7.2
matplotlib: 3.10.5
seaborn: 0.13.2
joblib: 1.5.2


## 2. Herramientas de desarrollo

In [3]:
import pytest
import nbformat
import nbclient

print('pytest:', pytest.__version__)
print('nbformat:', nbformat.__version__)
print('nbclient:', nbclient.__version__)

pytest: 9.1.1
nbformat: 5.11.1
nbclient: 0.11.0


In [4]:
import subprocess

try:
    git_version = subprocess.run(
        ['git', '--version'], capture_output=True, text=True, check=True
    ).stdout.strip()
    print('Git:', git_version)
except FileNotFoundError:
    print('Git: NO disponible')

Git: git version 2.49.0.windows.1


## 3. Módulo del proyecto (`src`)

In [5]:
import sys
from pathlib import Path

ROOT = Path.cwd().parent
sys.path.insert(0, str(ROOT / 'src'))

from inf8239_u01.environment import environment_message

mensaje = environment_message()
print('Mensaje del entorno:', mensaje)
assert mensaje == 'Entorno INF-8239 listo', 'El mensaje no coincide'

Mensaje del entorno: Entorno INF-8239 listo


## 4. Estructura del proyecto

In [6]:
from pathlib import Path

esperadas = [
    'data/raw', 'data/processed', 'docs', 'notebooks',
    'reports', 'src/inf8239_u01', 'tests',
    'README.md', 'requirements.txt', 'pyproject.toml',
]
for item in esperadas:
    existe = (ROOT / item).exists()
    print(f"{'OK ' if existe else 'FALTA'}  {item}")
    assert existe, f'Falta: {item}'

OK   data/raw
OK   data/processed
OK   docs
OK   notebooks
OK   reports
OK   src/inf8239_u01
OK   tests
OK   README.md
OK   requirements.txt
OK   pyproject.toml


## 5. Ejecución de la suite de tests

Ejecutar `pytest -q` desde la raíz del proyecto.
En el notebook se muestra una verificación ligera de que los tests existen y son descubribles.

In [7]:
import pytest

resultado = pytest.main(['-q', str(ROOT / 'tests')])
print('Código de salida de pytest:', resultado)
assert resultado == 0, 'Algun test falló'

.

.

.

.

.

.

.

.

.

.

.

.

.

.

.

.

.

.

                                                       [100%]

Código de salida de pytest: 0


## Conclusión del LAB00

El entorno (Python 3.13, NumPy, pandas, scikit-learn, matplotlib, seaborn, joblib, pytest, Jupyter, nbformat, nbclient y Git) está verificado y el módulo `inf8239_u01` responde correctamente.